tools/explorer/tt_adapter/src/tt_adapter/ttir.py

```diff
@@ -0,0 +1,186 @@
+            # ),
+            utils.make_editable_kv(
+                graph_builder.KeyValue(
+                    key="Memory Space",
+                    value=str(tt.MemorySpace(layout.memory_space_as_int)),
+                ),
+                editable={
+                    "input_type": "value_list",
+                    "options": utils.get_enum_options(tt.MemorySpace),
+                },
+            ),
+            utils.make_editable_kv(
+                graph_builder.KeyValue(
+                    key="Memory Layout",
+                    value=str(tt.TensorMemoryLayout(layout.memory_layout_as_int)),
+                ),
+                editable={
+                    "input_type": "value_list",
+                    "options": utils.get_enum_options(tt.TensorMemoryLayout),
+                },
+            ),
+            utils.make_editable_kv(
+                graph_builder.KeyValue(
+                    key="Grid Shape",
+                    value=array_ref_repr(layout.grid_attr.shape),
+                ),
+                editable={
+                    "input_type": "value_list",
+                    "options": ["1x1", "4x4", "8x8"],
+                },
+            ),
+        ]
+
+    # source_node.outputsMetadata.append(
+    #     graph_builder.MetadataItem(
+    #         id=str(connections[source_node.id]),
+    #         attrs=[
+    #             graph_builder.KeyValue(key="__tensor_tag", value=id)
+    #         ]
+    #         + source_node_attrs,
+    #     )
+    # )
+    # return []
+    return source_node_attrs
+
+    source_node.attrs.extend(source_node_attrs)
+
+
+def ttir_to_graph(module, ctx):
+    # Can assume that to-layout pass has already been run on the module.
+    name_dict = defaultdict(int)
+    output_connections = dict()
+    graph = graph_builder.Graph(id="ttir-graph")
+
+    op_to_graph_node = dict()
+
+    for op in module.body.operations:
+        append_later = []
+        for region in op.regions:
+            for block in region.blocks:
+                for op in block.operations:
+                    # Create all the nodes and constants in the first pass.
+                    graph_node = graph_builder.GraphNode(
+                        id=create_id(op, name_dict),
+                        label=op.name,
+                        namespace=create_namespace(op),
+                        attrs=get_attrs(op),
+                    )
+
+                    if op.name == "tensor.empty":
+                        append_later.append(graph_node)
+                    else:
+                        graph.nodes.append(graph_node)
+
+                    op_to_graph_node[op] = graph_node
+                    output_connections[graph_node.id] = 0
+
+                    for operand in op.operands:
+                        if operand.owner == block and operand not in op_to_graph_node:
+                            # This is a constant and we need to create a node for it.
```

Comment:
@vprajapati-tt While playing with the tool it seemed like it was very useful to have inputs / weights represented as visual nodes as well. What do you think?

Reply:
Yeah! This makes sense, I didn't do too much testing with graphs that had weights so this is reasonable.